dataset-processor-python/data_loader.py

```diff
@@ -1,55 +1,272 @@
+import json
 import os
+import re
 import sys
 import csv
+import pandas as pd
+import numpy
 
 
-class DataLoader:
+class UserInputHandler:
+    """
+    Class to handle user inputs
+    """
+
     def __init__(self, path, dataset_name):
         self.path = path
         self.dataset_name = dataset_name
-        self.initial_data = None
+        self.inputs = self.get_user_inputs()
+
+    @staticmethod
+    def is_valid_version(version):
+        pattern = r"^\d{4}\.[0-9]+$"
+        return re.match(pattern, version) is not None
+
+    def get_user_inputs(self):
+        version = input("Enter the version: ").strip()
+        if not version or not self.is_valid_version(version):
+            print("Please enter a valid version in the format [yyyy.number].")
+            return self.get_user_inputs()
+        title = input("Enter the title: ").strip()
+        description = input("Enter the description: ").strip()
+        return {
+            "path": self.path,
+            "dataset_name": self.dataset_name,
+            "version": version,
+            "title": title,
+            "description": description,
+        }
+
+
+class DataLoader:
+    """
+    Loads the csv file and stores the initial data
+    """
+
+    def __init__(self, inputs):
+        self.inputs = inputs
+        self.data = self.load_csv()
 
-    def read_csv(self):
+    def load_csv(self):
         """
-        read the csv file and store the data
+        Read the csv file and store the data
         """
         try:
-            with open(self.path, "r") as f:
+            with open(self.inputs["path"], "r") as f:
                 reader = csv.DictReader(f)
-                data = list(
-                    reader
-                )  # data is a list of dictionaries, each dictionary is a row
-                self.initial_data = data
+                return list(reader)
         except Exception as e:
             print(f"Error while reading CSV: {e}")
+            return None
 
 
-class DatasetWriter(DataLoader):
+class DatasetWriter:
     """
     Class to create the dataset folder and write json files
     """
 
-    def __init__(self, path, dataset_name):
-        super().__init__(path, dataset_name)
+    FILEINFO_COLUMN_NAMES = [
+        "cell_id",
+        "parent_id",
+        "cell_group",
+        "cell_thumbnail",
+        "cell_image",
+        "parent_thumbnail",
+        "parent_image",
+    ]
+
+    def __init__(self, data_loader, inputs):
+        # initialize the json files in the data folder
+        self.data = data_loader.data
+        self.inputs = inputs
+        self.cell_feature_analysis_data = []
+        self.feature_defs_data = []
+        self.dataset_data = {}
+        # utility variables - to organize and categorize the feature data for file writing
+        self.feature_def_keys = set()
+        self.discrete_features_dict = {}
+        self.features_data_order = []
```

Comment:
Left a comment on this before that I think is still unresolved. What is the key and value type for `self.discrete_features_dict`?